server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -466,13 +476,34 @@ public static void ensureMajorVersionBarrier(Version joiningNodeVersion, Version
         }
     }
 
+    public static void ensureNodeCommissioned(DiscoveryNode node, Metadata metadata) {
+        DecommissionAttributeMetadata decommissionAttributeMetadata = metadata.custom(DecommissionAttributeMetadata.TYPE);
+        if (decommissionAttributeMetadata != null) {
+            DecommissionAttribute decommissionAttribute = decommissionAttributeMetadata.decommissionAttribute();
+            DecommissionStatus status = decommissionAttributeMetadata.status();
+            if (decommissionAttribute != null && status != null) {
+                // We will let the node join the cluster if the current status is not IN_PROGRESS or SUCCESSFUL
+                if (node.getAttributes().get(decommissionAttribute.attributeName()).equals(decommissionAttribute.attributeValue())
+                    && (status.equals(DecommissionStatus.DECOMMISSION_IN_PROGRESS)
+                        || status.equals(DecommissionStatus.DECOMMISSION_SUCCESSFUL))) {
+                    throw new NodeDecommissionedException(
```

Comment:
Lets move this logic to DecommissionService?

Reply:
All other validators are present in this class itself and hence added it here. Also, it would be good if all the validators are present at one place. Let me know if this makes sense, else I can move it to service file